pkg/network/server.go

```diff
@@ -125,11 +125,14 @@ type (
 		// lastRequestedHeader contains a height of the last requested header.
 		lastRequestedHeader atomic.Uint32
 
-		register   chan Peer
-		unregister chan peerDrop
-		handshake  chan Peer
-		quit       chan struct{}
-		relayFin   chan struct{}
+		register       chan Peer
+		unregister     chan peerDrop
+		handshake      chan Peer
+		quit           chan struct{}
+		relayFin       chan struct{}
+		runFin         chan struct{}
+		broadcastTxFin chan struct{}
+		runProtoFin    chan struct{}
```

Comment:
Make commit title more descriptive. The main thing in this commit is that it fixes a bug in our network server that prevents server from proper shutdown and doesn't allow all server goroutines to finish normally just because the server exits without dependent goroutines awaiting. The consequence of this bug fix is that corresponding tests are fixed.